### DISCUSSION TITLE
Signing Signal Payloads

### DISCUSSION THREAD

**Discussion by isaacadams** [fennelLabs/Fennel-Protocol#39] [Ideas]
in part, we are playing around with the use of discussions, but this is also a serious discussion about #21 -- does every identity get a public/private key provided by the blockchain, which is then used to sign signal payloads?

**Comment by Romulus10** [comment]
MOST identities will GENERATE a keypair - this is not provided by the blockchain. I think these keys, however, will more commonly be used in chain-parallel communications than on-chain.

**Reply by isaacadams** [reply]
ok, you are saying the generation and management of keypairs will be off-chain?

**Reply by isaacadams** [reply]
side note. this is exactly what I was hoping the discussions world would look/feel like.

**Reply by Romulus10** [reply]
I am, yeah. Keypair generation is one of those classic tasks that would be a bad time for consensus. You also want to try the hardest you can to make sure entire private keys don't ever get generated remotely and sent over a network.

**Reply by Romulus10** [reply]
> side note. this is exactly what I was hoping the discussions world would look/feel like.

Me too 😂

**Comment by isaacadams** [comment]
based on this and the discussion in #21 -- it sounds like, for now, signal payloads will not be signed by an additional key.

**Reply by Romulus10** [reply]
I think it does make sense to leave it that way right now.
